src/wp-includes/comment-template.php

```diff
@@ -218,14 +218,50 @@ function get_comment_author_email_link( $linktext = '', $before = '', $after = '
  * @return string The comment author name or HTML link for author's URL.
  */
 function get_comment_author_link( $comment_ID = 0 ) {
+
 	$comment = get_comment( $comment_ID );
 	$url     = get_comment_author_url( $comment );
 	$author  = get_comment_author( $comment );
 
 	if ( empty( $url ) || 'http://' === $url ) {
 		$return = $author;
 	} else {
-		$return = "<a href='$url' rel='external nofollow ugc' class='url'>$author</a>";
+
+		$rel_parts = array();
+		if ( ! wp_is_internal_link( $url ) ) {
+			$rel_parts = array( 'external', 'nofollow' );
+		}
+
+		/**
+		 * Filters the list of non-ugc comment authors.
+		 *
+		 * When displaying the comment author link for any commenters in this list, the "ugc" rel attribute
+		 * will not be added to the anchor's rel attribute.
+		 *
+		 * @since [version]
+		 *
+		 * @param string[] $comment_authors An array of comment author usernames.
+		 */
+		$non_ugc_comment_authors = apply_filters( 'non_ugc_comment_authors', array() );
+		if ( ! in_array( $author, $non_ugc_comment_authors, true ) ) {
+			$rel_parts[] = 'ugc';
+		}
+
+		$rel = implode( ' ', $rel_parts );
+
+		/**
+		 * Filters the rel attribute of the comment author's link.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $rel        The rel attribute value.
+		 * @param string $author     The comment author's username.
+		 * @param string $comment_ID The comment ID as a numeric string.
+		 */
+		$rel = apply_filters( 'comment_author_link_rel', $rel, $author, $comment->comment_ID );
```

Comment:
With regards to `$rel` vs `$rel_parts` I think I basically copied from `make_clickable_rel`:

https://github.com/WordPress/wordpress-develop/blob/195393fdbbe798a5a8f4e1028c0f844766a65d40/src/wp-includes/formatting.php#L2926-L2935

I'd agree with you in general but was hesitant in case there was a good reason to follow this code. It probably doesn't constitute a common pattern that I should be bound to follow though. I'm just not intimately familiar with as much core code as regular contributors probably are.

Anyway, I'll update it since it's a new filter and it certainly is easier to work with the parts.

As par as the args, it's probably not even really necessary to pass anything other than the comment object, right?

So:

```
$rel_parts = apply_filters( '...', $rel_parts, $comment );
```

Where `$rel_part` is the array of parts to be imploded into the rel string and `$comment` is the `WP_Comment` object from `get_comment()`.